modules/ip/datapath/ip_input.c

```diff
@@ -26,6 +27,44 @@ enum edges {
 	EDGE_COUNT,
 };
 
+int format_ip_data(void *data, char *buf, size_t len) {
+	struct trace_ip_data *t = data;
+	const char *protos[UINT8_MAX] = {
+		[0] = "IPv6 Hop-by-Hop",
+		[1] = "ICMP",
+		[2] = "IGMP",
+		[4] = "IP in IP",
+		[6] = "TCP",
+		[17] = "UDP",
+		[27] = "Reliable Data Protocol",
+		[41] = "IPv6 Encapsulation",
+		[43] = "IPv6 Routing Header",
+		[44] = "IPv6 Fragment Header",
+		[47] = "GRE",
+		[50] = "ESP",
+		[51] = "Authentication Header",
+		[58] = "IPv6 ICMP",
+		[59] = "IPv6 No Next Header",
+		[60] = "IPv6 Destination Options",
+		[89] = "OSPF",
+		[112] = "VRRP",
+		[132] = "SCTP",
+		[136] = "UDP Lite",
+		[137] = "MPLS In IP",
+		[145] = "NSH",
+	};
+
+	return snprintf(
+		buf,
+		len,
+		IP4_ADDR_FMT " -> " IP4_ADDR_FMT " next proto: %s ttl %d",
+		IP4_ADDR_SPLIT(&t->src),
+		IP4_ADDR_SPLIT(&t->dst),
+		protos[t->proto],
```

Comment:
protos[t->proto] can be NULL

Reply:
yes, it safe to print, but we loose the proto id.
I'll fix that.